airbyte-webapp/src/views/Connector/ServiceForm/useAuthentication.tsx

```diff
@@ -0,0 +1,210 @@
+  const { values, getFieldMeta, submitCount } = useFormikContext<ServiceFormValues>();
+  const { selectedConnector } = useServiceForm();
+
+  const allowOAuthConnector = useFeature(FeatureItem.AllowOAuthConnector);
+
+  const advancedAuth = selectedConnector?.advancedAuth;
+  const legacyOauthSpec = selectedConnector?.authSpecification?.oauth2Specification;
+
+  const spec = selectedConnector?.connectionSpecification as JSONSchema7;
+
+  const isAuthButtonVisible = useMemo(() => {
+    const shouldShowAdvancedAuth =
+      advancedAuth && shouldShowButtonForAdvancedAuth(advancedAuth.predicateKey, advancedAuth.predicateValue, values);
+    const shouldShowLegacyAuth =
+      legacyOauthSpec && shouldShowButtonForLegacyAuth(spec, legacyOauthSpec.rootObject as Path, values);
+    return Boolean(allowOAuthConnector && (shouldShowAdvancedAuth || shouldShowLegacyAuth));
+  }, [values, advancedAuth, legacyOauthSpec, spec, allowOAuthConnector]);
+
+  // Fields that are filled by the OAuth flow and thus won't need to be shown in the UI if OAuth is available
+  const implicitAuthFieldPaths = useMemo(
+    () => [
+      // Fields from `advancedAuth` connectors
+      ...(advancedAuth
+        ? Object.values(serverProvidedOauthPaths(selectedConnector)).map((f) =>
+            makeConnectionConfigurationPath(f.path_in_connector_config)
+          )
+        : []),
+      // Fields from legacy `authSpecification` connectors
+      ...(legacyOauthSpec
+        ? [
+            ...convertAndPrefixPaths(legacyOauthSpec.oauthFlowInitParameters, legacyOauthSpec.rootObject as Path),
+            ...convertAndPrefixPaths(legacyOauthSpec.oauthFlowOutputParameters, legacyOauthSpec.rootObject as Path),
+          ]
+        : []),
+    ],
+    [advancedAuth, legacyOauthSpec, selectedConnector]
+  );
+
+  const isHiddenAuthField = useCallback(
+    (fieldPath: string) => {
+      // A field should be hidden due to OAuth if we have OAuth enabled and selected (in case it's inside a oneOf)
+      // and the field is part of the OAuth flow parameters.
+      return isAuthButtonVisible && implicitAuthFieldPaths.includes(fieldPath);
+    },
+    [implicitAuthFieldPaths, isAuthButtonVisible]
+  );
+
+  const hiddenAuthFieldErrors = useMemo(() => {
+    if (!isAuthButtonVisible) {
+      // We don't want to return the errors if the auth button isn't visible.
+      return {};
+    }
+    return implicitAuthFieldPaths.reduce<Record<string, string>>((authErrors, fieldName) => {
+      const { error } = getFieldMeta(fieldName);
+      if (submitCount > 0 && error) {
+        authErrors[fieldName] = error;
+      }
+      return authErrors;
+    }, {});
+  }, [getFieldMeta, implicitAuthFieldPaths, isAuthButtonVisible, submitCount]);
+
+  const shouldShowAuthButton = useCallback(
+    (fieldPath: string) => {
+      if (!isAuthButtonVisible) {
+        // Never show the auth button anywhere if its not enabled or visible (inside a conditional that's not selected)
```

Comment:
should this say enabled AND visible?

Reply:
could be :D Not a native speaker, I thought in a negated sentence `or` described that neither of both should be true :D